app/src/main/java/com/shevchenkovtwo/homework/MovieDetailFragment.kt

```diff
@@ -26,38 +28,52 @@ class MovieDetailFragment : Fragment() {
     }
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
-        fragmentMovieDetailBinding?.let {
-            it.apply {
-                movieName.text = selectedMovie?.name
-                movieRating.rating = selectedMovie!!.rating
-                selectedMovie!!.storyline?.let { storyline -> movieStoryline.setText(storyline) }
-                pg.text = selectedMovie?.pg
-                reviews.text = selectedMovie?.reviews
-                tag.text = selectedMovie?.tags
-                selectedMovie!!.imageMask?.let { image -> mask.setImageResource(image) }
-                if (selectedMovie!!.favorite) {
-                    favourite.setImageResource(R.drawable.ic_favorite)
-                } else {
-                    favourite.setImageResource(R.drawable.ic_not_favorite)
+        super.onViewCreated(view, savedInstanceState)
+        movieDetailViewModel = ViewModelProvider(this).get(MovieDetailViewModel::class.java)
+        movieDetailViewModel?.let { movieViewModel ->
+            movieViewModel.getSelectedMovie().observe(viewLifecycleOwner, { movie ->
+                fragmentMovieDetailBinding?.let { binding ->
+                    initView(movie, binding)
+                    navigateBack(binding)
```

Comment:
This is not really connected to selected movie, so you don't need to do it every time when new movie is selected